### PR TITLE
implement global cors configuration

### PR DESCRIPTION
**Pull request by tharindu1st**
implement global cors configuration

### CODE REVIEW

File: adapter/config/default_config.go [hunk 1]
```diff
@@ -175,6 +175,15 @@ var defaultConfig = &Config{
 				EnableOutboundCertificateHeader: false,
 			},
 		},
+		Cors: cors{
+			Enabled:                       true,
```
Comment (AmaliMatharaarachchi) [inline adapter/config/default_config.go:179]: are we enabling cors default for all apis? I rem like we disable cors by default. (enbling cors for unnecesary apis will add a computational overhead)

File: adapter/internal/operator/controllers/dp/api_controller.go [hunk 2]
```diff
@@ -1072,6 +1072,7 @@ func (apiReconciler *APIReconciler) retriveParentAPIsAndUpdateOwnerReferene(ctx
 			return
 		}
 		requests = apiReconciler.getAPIsForAPIPolicy(ctx, &apiPolicy)
+		loggers.LoggerAPKOperator.Infof("API Policy requests %v", apiPolicy)
```
Comment (AmaliMatharaarachchi) [inline adapter/internal/operator/controllers/dp/api_controller.go:1075]: is this intentional?

File: common-go-libs/apis/dp/v1alpha1/authentication_conversion.go [hunk 3]
```diff
@@ -31,65 +31,72 @@ func (src *Authentication) ConvertTo(dstRaw conversion.Hub) error {
...
+			// Convert Oauth2Auth Default to v1alpha2.APIKey : Required field added as optional for APIKey
+			for _, apiKeyAuth := range defaultAuthv1Spec.AuthTypes.APIKey {
+				convertedAPIKeyAuth := v1alpha2.APIKeyAuth{
+					In:                  apiKeyAuth.In,
+					Name:                apiKeyAuth.Name,
+					SendTokenToUpstream: apiKeyAuth.SendTokenToUpstream,
+				}
+				v1alpha2authTypes.APIKey = append(v1alpha2authTypes.APIKey, convertedAPIKeyAuth)
+			}
+			if defaultAuthv1Spec.AuthTypes.TestConsoleKey != (TestConsoleKeyAuth{}) {
```
Comment (AmaliMatharaarachchi) [inline common-go-libs/apis/dp/v1alpha1/authentication_conversion.go:55]: does this != work when it empty struct is given? I'm not sure golang assign same address for empty object and != checks for address right?

File: common-go-libs/apis/dp/v1alpha1/authentication_webhook.go [hunk 4]
```diff
@@ -18,14 +18,7 @@
 package v1alpha1
 
 import (
-	constants "github.com/wso2/apk/common-go-libs/constants"
```
Comment (AmaliMatharaarachchi) [inline common-go-libs/apis/dp/v1alpha1/authentication_webhook.go:21]: we can remove the whole file

File: common-go-libs/apis/dp/v1alpha2/apipolicy_conversion.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (AmaliMatharaarachchi) [inline common-go-libs/apis/dp/v1alpha2/apipolicy_conversion.go:2]:
```suggestion
 *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: adapter/config/default_config.go [hunk 6]
```diff
@@ -175,6 +175,15 @@ var defaultConfig = &Config{
...
+			AccessControlAllowOrigins:     []string{"*"},
+			AccessControlAllowMethods:     []string{"GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS"},
+			AccessControlAllowHeaders:     []string{"authorization", "Access-Control-Allow-Origin", "Content-Type"},
```
Comment (pubudu538) [inline adapter/config/default_config.go:182]: Add Internal-key

File: adapter/config/default_config.go (hunk 1 above)
Comment (tharindu1st) [reply adapter/config/default_config.go:179]: this applies when v1 to v2 migration of apipolicy.

File: adapter/internal/operator/controllers/dp/api_controller.go (hunk 2 above)
Comment (tharindu1st) [reply adapter/internal/operator/controllers/dp/api_controller.go:1075]: can remove

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**